src/app/theme/components/baD3Chart/baD3Chart.component.ts

```diff
@@ -0,0 +1,63 @@
+import {
+    Component,
+    ViewChild,
+    ViewEncapsulation,
+    Input,
+    Output,
+    EventEmitter,
+    AfterViewInit,
+    ElementRef
+} from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/distinctUntilChanged';
+
+import { BaD3ChartService } from './baD3Chart.service';
+import { d3 } from './baD3Chart.loader';
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'ba-d3-chart',
+    template: require('./baD3Chart.html'),
+    styles: [require('./baD3Chart.scss')],
+    providers: [BaD3ChartService]
+})
+export class BaD3Chart{
+    @Input()  baD3ChartType:string;
+    @Input()  baD3ChartData:any;
+    @Output() onChartReady = new EventEmitter<any>();
+    @ViewChild('baD3Chart') private _selector:ElementRef;
+
+    constructor(private _chartService:BaD3ChartService){
+    }
+
+    setup(){
+        this._chartService.chartSetup = {
```

Comment:
I would say it's better to pass this configuration directly into the draw method of a specific chart type as in this case we won't save any state inside the service making it more-or-less stateless.